### ISSUE TITLE
Add ignoreFunctions: [] to unit-allowed-list

### ISSUE THREAD

**Issue by treemmett** [stylelint/stylelint#5851]
### What steps are needed to reproduce the bug?

Use css/scss functions that include a disallowed unit character.

```scss
button {
  border: 1px solid hsl(20deg 0% 20% / 80%);
  background-color: lighten(green, 20%);
}

```

### What Stylelint configuration is needed to reproduce the bug?

```json
{
  "rules": {
    "unit-allowed-list": "px"
  }
}

```

### How did you run Stylelint?

CLI `stylelint **/*.scss`

### Which version of Stylelint are you using?

14.2.0

### What did you expect to happen?

Units within function to be ignored.

### What actually happened?

The following problems were reported

```
2:25 errorUnexpected unit "deg" (unit-allowed-list)
2:31 errorUnexpected unit "%" (unit-allowed-list)
2:34 errorUnexpected unit "%" (unit-allowed-list)
2:40 errorUnexpected unit "%" (unit-allowed-list)
3:36 errorUnexpected unit "%" (unit-allowed-list)
```

### Does the bug relate to non-standard syntax?

Yes, SCSS color functions and vanilla HSL functions

### Proposal to fix the bug

Add `ignore: ["inside-function"]` secondary option to unit-allowed-list, similar fix to https://github.com/stylelint/stylelint/issues/5140

**Comment by ybiquitous** [comment]
@treemmett Thank you for the report, using the template, and providing the nice demo! 😄 

Your proposal to add `ignore: ["inside-function"]` option sounds good to me.

What do you think, @jeddy3?

**Comment by jeddy3** [comment]
A new option sounds good.

I'm wondering whether we missed an opportunity when we added `ignore: ["inside-function"]` to those other rules. It's too broad. If we pivot to a [user defined list option](https://stylelint.io/user-guide/rules/about#user-defined-ignore), i.e. `ignoreFunction: []` we can cater to the above use case and others.

For example, given:

```json
{
  "rules": {
    "unit-allowed-list": ["px", { "ignoreFunctions": ["hsl"] }]
  }
}
```

Would allow:

```css
a {
  border: 1px solid hsl(20deg 0% 20% / 80%);
}
```

But disallow:

```css
a {
  font-size: 10px;;
}

b {
  font-size: var(--font-size, 1rem);
}
```

(When you want font sizes always in pixels).

You can still ignore all functions using this option by using regex in the list.

Thoughts?

**Comment by ybiquitous** [comment]
Great idea! 👍🏼

**Comment by jeddy3** [comment]
Okay, let's go with `ignoreFunctions: []`.

I've labelled the issue as ready to implement. @treemmett Please consider [contributing](https://stylelint.io/contributing) if you have time.

There are [steps on how to add a new option](https://stylelint.io/developer-guide/rules#add-an-option-to-a-rule) in the Developer guide.
